ReSharper.FSharp/src/FSharp.Psi.Features/src/Daemon/Analyzers/LambdaAnalyzer.fs

```diff
@@ -129,7 +129,7 @@ type LambdaAnalyzer() =
                 if compareArg pat expr then LambdaCanBeReplacedWithBuiltinFunctionWarning(lambda, "id") :> _ else
 
                 match pat with
-                | :? ITuplePat as pat when pat.PatternsEnumerable.CountIs(2) ->
+                | :? ITuplePat as pat when isNull pat.StructKeyword && pat.PatternsEnumerable.CountIs(2) ->
```

Comment:
Could we have it encapsulated as something like `ITuplePat.IsStruct` and use it here instead?